utils/build-script

```diff
@@ -1040,455 +1513,37 @@ details of the setups of other systems or automated environments.""")
     if args.cmake is not None:
         toolchain.cmake = args.cmake
 
-    if toolchain.cc is None or toolchain.cxx is None:
-        diagnostics.fatal(
-            "can't find clang (please install clang-3.5 or a later version)")
+    # Validate the arguments.
+    BuildScriptInvocation.validate_arguments(toolchain, args)
 
-    if toolchain.cmake is None:
-        diagnostics.fatal("can't find CMake (please install CMake)")
+    # Preprocess the arguments to apply defaults.
+    BuildScriptInvocation.apply_default_arguments(toolchain, args)
```

Comment:
@ddunbar I was just working on something and I noticed that default arguments (including propagation) happens after validation. Shouldn't it happen the other way around?

For example, if you specify and 'android' target in your `stdlib-deployment-targets`, that should imply `--android` because it does the same thing (tries to configure). In that case, we want to go through the same validation routines, checking for NDK paths, etc. It makes more sense to me that this propagation happens in `apply_default_arguments` instead of in the validation function.